shell/browser/native_window_views.cc

```diff
@@ -1525,6 +1525,13 @@ void NativeWindowViews::SetVisibleOnAllWorkspaces(
 }
 
 bool NativeWindowViews::IsVisibleOnAllWorkspaces() const {
+  // NB: Electron >= 37 has a better long-term fix, but it also has an edge
+  // case which is a breaking change. The code here is dirtier (e.g. accessing
+  // a method marked as private) to avoid that edge case. More info @
+  // https://github.com/electron/electron/pull/46834#issuecomment-2836287699
+  if (const auto* view_native_widget = widget()->native_widget_private())
+    return view_native_widget->IsVisibleOnAllWorkspaces();
```

Comment:
Electron 37 delegates this out to [Widget::IsVisibleOnAllWorkspaces()](https://chromium.googlesource.com/chromium/src/+/main/ui/views/widget/widget.cc#1114).

What this code does is copy the logic of that function -- returning `true` if `widget() && widget()->IsVisibleOnAllWorkspaces()` -- and then falls back to the old implementation otherwise. Keeping the old implementation should prevent the breaking change in 36.